src/main/java/htsjdk/samtools/SAMTextReader.java

```diff
@@ -221,7 +221,12 @@ private RecordIterator() {
         }
 
         public void close() {
-            SAMTextReader.this.close();
```

Comment:
I think this will end up with weird situations still.  I.e. we close the iterator but calling hasNext() still returns true